core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ApplyPreferredTableExecutePartitioning.java

```diff
@@ -58,11 +61,15 @@ public Result apply(TableExecuteNode node, Captures captures, Context context)
             return enable(node);
         }
 
-        double expectedNumberOfPartitions = getRowsCount(
-                context.getStatsProvider().getStats(node.getSource()),
-                node.getPreferredPartitioningScheme().get().getPartitioning().getColumns());
+        PlanNodeStatsEstimate stats = context.getStatsProvider().getStats(node.getSource());
+        Set<Symbol> partitionColumns = node.getPreferredPartitioningScheme().get().getPartitioning().getColumns();
+        if (!canEstimateNumberOfPartitions(stats, partitionColumns)) {
```

Comment:
We don't have to since in the above check if there's a nan we are simply enabling it.

Reply:
Change this:
```
if (!isNan(expectedNumberOfPartitions) && expectedNumberOfPartitions < minimumNumberOfPartitions) { return Result.empty(); }
```

into:
```
if (isNan(expectedNumberOfPartitions)) {
   // Force (...)
   return enable(node): 
}

if (expectedNumberOfPartitions < minimumNumberOfPartitions) { 
   return Result.empty(); 
}
```
?

Reply:
No, my point is that `expectedNumberOfPartitions` can never be nan now right?

Reply:
Oh sorry, I got the question wrong. We need to check whether statistics exist or not using `canEstimateNumberOfPartitions`. That's because `getRowsCount(stats, partitionColumns)` can return 0 or 1 even when `distinctValuesCount` for columns doesn't exist.

Reply:
> That's because getRowsCount(stats, partitionColumns) can return 0 or 1 when distinctValuesCount for grouping columns is equal to 0.

Why is that a problem? It means `getRowsCount` could estimate row count even though partial stats are missing, which is fine.